$OpenBSD: patch-meson_build,v 1.8 2021/10/03 19:53:21 ajacoutot Exp $

-fexceptions:
ld: error: undefined symbol: _Unwind_Resume

Should match b_lundef in meson(1).

Index: meson.build
--- meson.build.orig
+++ meson.build
@@ -234,7 +234,6 @@ array_bounds = get_option('b_sanitize') == 'none' ? 2 
 
 cc_flags += [
   '-fasynchronous-unwind-tables',
-  '-fexceptions',
   '-fipa-pure-const',
   '-fno-common',
   '-Wabsolute-value',
@@ -497,11 +496,6 @@ libvirt_nodelete = cc.get_supported_link_arguments([
 ])
 
 libvirt_no_undefined = []
-if get_option('b_sanitize') == 'none'
-  libvirt_no_undefined += cc.get_supported_link_arguments([
-    '-Wl,-z,defs',
-  ])
-endif
 
 libvirt_no_indirect = cc.get_supported_link_arguments([
   '-Wl,--no-copy-dt-needed-entries',
@@ -949,12 +943,15 @@ if devmapper_dep.found()
   conf.set('WITH_DEVMAPPER', 1)
 endif
 
-dlopen_use = host_machine.system() != 'windows'
+dlopen_use = host_machine.system() != 'windows' and host_machine.system() != 'openbsd'
 dlopen_dep = cc.find_library('dl', required: dlopen_use)
 if dlopen_dep.found()
   if not cc.has_header('dlfcn.h')
     error('Unable to find dlfcn.h')
   endif
+  conf.set('WITH_DLFCN_H', 1)
+endif
+if host_machine.system() == 'openbsd'
   conf.set('WITH_DLFCN_H', 1)
 endif
 
